evaluator/src/main/resources/rules/osadl.rules.kts

```diff
@@ -0,0 +1,114 @@
+/*
+ * Copyright (C) 2022 The ORT Project Authors (see <https://github.com/oss-review-toolkit/ort/blob/main/NOTICE>)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ * License-Filename: LICENSE
+ */
+
+// Use this rule like:
+//
+// $ ort evaluate -i scanner/src/funTest/assets/dummy-expected-output-for-analyzer-result.yml --rules-resource /rules/osadl.rules.kts
+
+enum class Compatibility {
+    INHERENT,
+    YES,
+    NO,
+    UNKNOWN;
+
+    companion object {
+        fun fromString(value: String): Compatibility =
+            when (value) {
+                "", "Same" -> INHERENT
+                "Yes" -> YES
+                "No" -> NO
+                "?", "Unknown", "Check dependency", "Dep." -> UNKNOWN
+                else -> throw IllegalArgumentException("Unhandled compatibility '$value' found.")
+            }
+    }
+}
+
+val matrix = mutableMapOf<Pair<String, String>, Compatibility>()
+
+javaClass.getResourceAsStream("/rules/osadl-matrix.csv").bufferedReader().useLines { lines ->
+    val rowIterator = lines.iterator()
+
+    // The first row contains the header with names of the inbound licenses.
+    val columnHeaders = rowIterator.next().split(',')
+        // Drop the surrounding "Compatibility*" cells from the header row.
+        .drop(1).dropLast(1)
+
+    while (rowIterator.hasNext()) {
+        val cellIterator = rowIterator.next().split(',').iterator()
+
+        // The first column contains the header with names of the outbound licenses.
+        val inbound = cellIterator.next()
+
+        columnHeaders.forEach { outbound ->
+            val value = cellIterator.next()
+            matrix[inbound to outbound] = Compatibility.fromString(value)
+        }
+    }
+}
+
+fun PackageRule.LicenseRule.isOsadlMatrixNonCompliant(inbound: SpdxSingleLicenseExpression) =
```

Comment:
I would rather name this `isOsadlMatrixCompliant` and use the `-` operator in the `require` block.